flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/filesystem/AbstractFileSystemTable.java

```diff
@@ -59,15 +68,19 @@ ReadableConfig formatOptions(String identifier) {
         return new DelegatingConfiguration(tableOptions, identifier + ".");
     }
 
-    DataType getFormatDataType() {
-        TableSchema.Builder builder = TableSchema.builder();
-        schema.getTableColumns()
-                .forEach(
-                        column -> {
-                            if (!partitionKeys.contains(column.getName())) {
-                                builder.add(column);
-                            }
-                        });
-        return builder.build().toRowDataType();
+    DataType getRowDataTypeWithoutPartitionColumns() {
```

Comment:
This can be entirely derived from `getRowDataType` by using `getRowDataType().getLogicalType()` which is always `RowType` and has all the information.

Reply:
TBH I find less convoluted the committed code than extracting the info from `getRowDataType()`:

```java
FieldsDataType fieldsDataType = (FieldsDataType) getRowDataType();
RowType rowType = (RowType) getRowDataType().getLogicalType();
return ROW(
        IntStream.range(0, rowType.getFieldCount())
                .filter(i -> !partitionKeys.contains(rowType.getFieldNames().get(i)))
                .mapToObj(i -> DataTypes.FIELD(rowType.getFieldNames().get(i), fieldsDataType.getChildren().get(i)))
                .toArray(DataTypes.Field[]::new)
);
```

Same for the comment in `DeserializationSchemaAdapter`.

Reply:
Oh wait I've found some utils to make this code look cleaner, let me refactor it